### DISCUSSION TITLE
Matching bracket only highlighted for primary with multi cursor

### DISCUSSION THREAD

**Discussion by A-Walrus** [helix-editor/helix#10413] [General]
When using multiple cursors (like with `C`) with the cursor over brackets, the matching bracket is only highlighted for the primary selection.
![image](https://user-images.githubusercontent.com/58790821/181520278-71c53622-dc9c-4d82-8d94-3ead3dc4a3f4.png)

What do you think?
- This is intended behavior and shouldn't change
- All matching brackets should always be highlighted
- This should be configurable, 3 options: `all, primary, none` (what should be the default)?

**Comment by nathan-at-least** [comment]
Tangent on configuring the matching indicator shape:

@A-Walrus how are you able to get the matching indicator to be an underline while the cursor is a block?

In my editor right now this is how matching renders:
![image](https://github.com/helix-editor/helix/assets/4369700/c29216f7-3267-409e-9fc3-956a28b63c64)

The cursor is grey while the matching indicator is orange. Both are blocks. For me, distinguishing by color alone is really error prone, so I often get lost when navigating around brackets as to where the actual cursor is. I'd love to configure the match indicator to be an underscore as in your rendering.

I have found I can reconfigure the cursor with `editor.cursor-shape.normal` and I can set that to a bar or underline. The problem is I want the main cursor to be a block and the matching indicator to be a bar.

I searched issues, PRs, and discussions for this issue and couldn't find it, although [this discussion](https://github.com/helix-editor/helix/discussions/4833) gets close (it is about colors only IIUC).

**Comment by A-Walrus** [comment]
@nathan-at-least This is up to the theme. Defined by the `ui.cursor.match` key. You can try modifying your theme (or creating a new theme that inherits from it) to add it. The screenshot is with the `dark_plus` theme
